.github/workflows/all-static-checks.yaml

```diff
@@ -61,3 +61,62 @@ jobs:
         run: |
           pip install pyyaml
           python tests/sweep_framework/sweeps_workflow_verification.py
+    name: Static analysis
+  clang-tidy:
+    runs-on: ubuntu-latest
+    container: ghcr.io/tenstorrent/tt-metal/tt-metalium/ubuntu-22.04-amd64:latest
+    permissions:
+      pull-requests: write
+      # OPTIONAL: auto-closing conversations requires the `contents` permission
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+        submodules: "recursive"
+    - name: Set safe directory for Git
+      run: git config --global --add safe.directory /__w/tt-metal/tt-metal
+    - name: Fetch base branch
+      run: |
+        git remote add upstream "https://github.com/${{ github.event.pull_request.base.repo.full_name }}"
+        git fetch --no-tags upstream "${{ github.event.pull_request.base.ref }}"
+      continue-on-error: true
```

Comment:
If we fail here, shouldn't we bail out so we become aware and can go fix it?

Reply:
My initial instinct was, I never want clang-tidy to fail a PR, but you are right that if we bypass errors, we will never know that its broken. Sounds like your vote is to allow it to error out.